sw/airborne/modules/nav/nav_survey_poly_rotorcraft.c

```diff
@@ -155,7 +155,7 @@ bool nav_survey_poly_setup(uint8_t EntryWP, uint8_t Size, float sw, float Orient
   CSurveyStatus = Init;
 
   if (Size == 0) {
-    return true;
+    return;
```

Comment:
I updated the run function to return without doing anything if the size is zero...